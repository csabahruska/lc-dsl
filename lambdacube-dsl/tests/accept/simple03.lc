
renderWire fb = let
  blend = Blend (FuncSubtract,FuncAdd) ((SrcColor,SrcColor),(SrcColor,OneMinusSrcAlpha)) (V4F 0.0 1.0 0.0 0.0)
  modelViewProj = Uni (IM44F "viewProj")
  polyMode          = PolygonLine 20.0
  polyMode'        = PolygonPoint (PointSize 10.0)
  polyMode''         = PolygonFill
  rasterCtx         = TriangleCtx CullNone polyMode (Offset (-1) 0) FirstVertex
  fragmentCtx       = AccumulationContext (DepthOp Lequal True,ColorOp NoBlending (V4B True True True True))
  vertexShader (v,n)    = VertexOut (PrimMulMatVec modelViewProj (PrimV3FToV4F v) ) 1.0 () (Smooth (PrimV3FToV4F n))
  vertexStream      = Fetch "missing shader" Triangles (IV3F "position",IV3F "color")
  primitiveStream   = Transform vertexShader vertexStream
  fragmentStream    = Rasterize rasterCtx primitiveStream
  fragmentShader f  = FragmentOutRastDepth (V4F 1.0 1.0 1.0 1.0)
 in Accumulate fragmentCtx PassAll fragmentShader fragmentStream fb

render fb = let
  blend = Blend (FuncSubtract,FuncAdd) ((SrcColor,SrcColor),(SrcColor,OneMinusSrcAlpha)) (V4F 0.0 1.0 0.0 0.0)
  modelViewProj = Uni (IM44F "viewProj")
  polyMode          = PolygonLine 20.0
  polyMode'        = PolygonPoint (PointSize 10.0)
  polyMode''         = PolygonFill
  rasterCtx         = TriangleCtx CullNone polyMode'' NoOffset FirstVertex
  fragmentCtx       = AccumulationContext (DepthOp Less True,ColorOp NoBlending (V4B True True True True))
  vertexShader (v,n)    = VertexOut (PrimMulMatVec modelViewProj (PrimV3FToV4F v) ) 1.0 () (Smooth (PrimV3FToV4F n))
  vertexStream      = Fetch "missing shader" Triangles (IV3F "position",IV3F "color")
  primitiveStream   = Transform vertexShader vertexStream
  fragmentStream    = Rasterize rasterCtx primitiveStream
  fragmentShader f  = FragmentOutRastDepth f-- (V4F 1.0 1.0 1.0 0.2)
 in Accumulate fragmentCtx PassAll fragmentShader fragmentStream fb

main = let
  emptyFB = FrameBuffer (DepthImage @1 1000,ColorImage @1 (V4F 0.0 0.0 0.4 1.0))
 in ScreenOut $ renderWire $ render emptyFB
