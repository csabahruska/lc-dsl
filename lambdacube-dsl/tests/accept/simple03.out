Pipeline {backend = OpenGL33, textures = [], samplers = [], targets = [RenderTarget {renderTargets = [(Depth,Just (Framebuffer Depth)),(Color,Just (Framebuffer Color))]}], programs = [Program {programUniforms = fromList [("viewProj",M44F)], programStreams = fromList [("n",("color",V3F)),("v",("position",V3F))], programInTextures = fromList [], programOutput = [("f0",V4F)], vertexShader = "#version 330 core\nuniform mat4 viewProj ;\nin vec3 v ;\nin vec3 n ;\nsmooth out vec4 v0 ;\nvoid main() {\nv0 = vec4( n ,1.0);\ngl_Position = ( viewProj ) * ( vec4( v ,1.0) );\ngl_PointSize = 1.0;\n}\n", geometryShader = Nothing, fragmentShader = "#version 330 core\nsmooth in vec4 v0 ;\nout vec4 f0 ;\nvoid main() {\nf0 = vec4 ( 1.0,1.0,1.0,1.0 );\n}\n"},Program {programUniforms = fromList [("viewProj",M44F)], programStreams = fromList [("n",("color",V3F)),("v",("position",V3F))], programInTextures = fromList [], programOutput = [("f0",V4F)], vertexShader = "#version 330 core\nuniform mat4 viewProj ;\nin vec3 v ;\nin vec3 n ;\nsmooth out vec4 v0 ;\nvoid main() {\nv0 = vec4( n ,1.0);\ngl_Position = ( viewProj ) * ( vec4( v ,1.0) );\ngl_PointSize = 1.0;\n}\n", geometryShader = Nothing, fragmentShader = "#version 330 core\nsmooth in vec4 v0 ;\nout vec4 f0 ;\nvoid main() {\nf0 = v0;\n}\n"}], slots = [Slot {slotName = "missing shader", slotUniforms = fromList [("viewProj",M44F)], slotStreams = fromList [("color",V3F),("position",V3F)], slotPrimitive = Triangles, slotPrograms = [0,1]}], commands = [SetRenderTarget 0,ClearRenderTarget [(Depth,VFloat 1000.0),(Color,VV4F (V4 0.0 0.0 0.4 1.0))],SetRasterContext (TriangleCtx CullNone PolygonFill NoOffset FirstVertex),SetAccumulationContext (AccumulationContext {accViewportName = Nothing, accOperations = [DepthOp Less True,ColorOp NoBlending (VV4B (V4 True True True True))]}),SetProgram 1,RenderSlot 0,SetRasterContext (TriangleCtx CullNone (PolygonLine 20.0) (Offset (-1.0) 0.0) FirstVertex),SetAccumulationContext (AccumulationContext {accViewportName = Nothing, accOperations = [DepthOp Lequal True,ColorOp NoBlending (VV4B (V4 True True True True))]}),SetProgram 0,RenderSlot 0]}